cli/src/commands/status.rs

```diff
@@ -162,6 +164,21 @@ pub(crate) fn cmd_status(
                 }
             }
         }
+        let unmatched_paths =
+            print_unmatched_explicit_paths(ui, &workspace_command, &fileset_expression, [&tree])?;
+
+        if let [only_path] = args.paths.as_slice() {
+            if !unmatched_paths.is_empty()
+                && workspace_command
+                    .parse_revset(ui, &RevisionArg::from(only_path.to_owned()))
+                    .is_ok()
+            {
+                writeln!(
+                    ui.hint_default(),
+                    "Maybe you meant `jj diff --summary -r {only_path:?}`?",
```

Comment:
nit: "To show changes in a certain revision, use `jj diff ...`" or something? It's not so obvious that the user expects `jj status` could show changes of arbitrary revision.